### PR TITLE
Make sure Settings also work when running on Plasma

### PR DESCRIPTION
**Pull request by aleixpol**
Moves the call into a C++ class.
Checks the system being run using the XDG_CURRENT_DESKTOP.
Prevents us from forwarding the QT_WAYLAND_SHELL_INTEGRATION value to
the child process, since it will not work as expected as we are
enforcing a private one. Made the switch for all desktops as I
understand it's the same problem elsewhere.

Note, this will only work properly as of Plasma 5.23 (October). The current case is already broken so I'd say we can safely get this in and eventually have it all converge.

### CODE REVIEW

**Comment by KAMiKAZOW** [comment]
@jpetersen This PR is a no-brainer. Please accept. If you're too busy looking at PRs, I'll happily click on the Merge button if you give me the required permissions...

File: src/plugin/inputmethod.cpp [hunk 1]
```diff
@@ -755,3 +755,19 @@ void InputMethod::onPluginPathsChanged(const QStringList& pluginPaths) {
 
     d->updateLanguagesPaths();
 }
+
+void InputMethod::showSystemSettings()
+{
+    auto previous = qgetenv("QT_WAYLAND_SHELL_INTEGRATION");
+    qunsetenv("QT_WAYLAND_SHELL_INTEGRATION");
```
Comment (dobey) [inline src/plugin/inputmethod.cpp:762]: Why is this necessary? A comment to explain would be nice, as it's not intuitive as to why this would be needed.

File: src/plugin/inputmethod.cpp [hunk 2]
```diff
@@ -755,3 +755,19 @@ void InputMethod::onPluginPathsChanged(const QStringList& pluginPaths) {
...
+
+    if (qgetenv("XDG_CURRENT_DESKTOP") == "KDE") {
+        QDesktopServices::openUrl(QUrl("systemsettings://kcm_virtualkeyboard"));
+    } else {
+        QDesktopServices::openUrl(QUrl("settings://system/language"));
+    }
```
Comment (dobey) [inline src/plugin/inputmethod.cpp:768]: Do we have a kcm yet for configuring maliit? Should we provide instead a more generic way of configuring things that can work on all environments? As I understand, neither of these URLs will do anything useful for maliit-keyboard yet anyway.

**Review by dobey — CHANGES_REQUESTED** [review]

File: src/plugin/inputmethod.cpp (hunk 1 above)
Comment (aleixpol) [reply src/plugin/inputmethod.cpp:762]: Addressed.

**Comment by bhush9** [comment]
This change on its own looks good to me, but I wonder if it makes sense to put "Launch system settings" command in the gsettings so it can be easily overridden?

**Comment by aleixpol** [comment]
> This change on its own looks good to me, but I wonder if it makes sense to put "Launch system settings" command in the gsettings so it can be easily overridden?

We can abstract anything further. I'd recommend changing this as is and considering any further abstraction as an iteration to the problem.

**Review by dobey — APPROVED** [review]
Looks OK. Further refactoring can be done later, but would be nice to have this opening the settings properly on Plasma Mobile. Perhaps we should instead get rid of the settings plug-ins for the future, and just have the keyboard settings be a hidden app provided by maliit keyboard? Would make things simpler for using the keyboard across multiple environments at least.
